pulsar/consumer_impl.go

```diff
@@ -283,8 +284,9 @@ func (c *consumer) Nack(msg Message) {
 }
 
 func (c *consumer) NackID(msgID MessageID) {
-	mid, ok := c.messageID(msgID)
-	if !ok {
+	mid, err := newMessageIDFromInterface(msgID, len(c.consumers))
```

Comment:
I'm confused what is the missing error?

Reply:
if the `msgID` interface paramter is not of the internal type `messageID` `newMessageIDFromInterface()` will fail

Reply:
Why is that error needed? It looks like it's just being logged and not returned to the client?